views/widget.go

```diff
@@ -108,7 +117,13 @@ func (ww *WidgetWatchers) Unwatch(handler tcell.EventHandler) {
 // PostEvent delivers the EventWidget to all registered watchers.  It is
 // to be called by the Widget implementation.
 func (ww *WidgetWatchers) PostEvent(wev EventWidget) {
-	for watcher := range ww.watchers {
```

Comment:
I am going to approve this for now, but I feel like there is a race here that should be called out ...

if a user removes a watcher, we might be trying to issue a change, and I feel like there is a subtle where the watcher can get events after it is removed.  But honestly, this change is an improvement over the thread-unsafe stuff from before -- so its better to take it as is and deal with that race in the unlikely event that anyone complains about it later.

Reply:
I agree, that condition probably isn't great to have. But it's at least something that downstream can handle on their own to prevent the race. Something like this:

```go
func someFunc() {
	childWidget.Unwatch(myWidget)
	myWidget.Lock()
	myWidget.watchingChildWidget = false
	myWidget.Unlock()
}

...

func (m *myWidget) HandleEvent(ev tcell.Event) error {
	m.Lock()
	defer m.Unlock()
	if !myWidget.watchingChildWidget {
		return someError
	}
	// handle the event
}
```

Thanks for merging!